specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,568 @@
+
+Load logic depends on the [technology] in use.
+It is the mechanism by which relevant verifier information from all load paths is retrieved, while considering the [os], [purpose], [context] and [technology] request of a calling application.
+
+For some technologies, the first suitable _signature verifier_ found in the list of [load paths] is used, effectively [overriding] files with the same name in directories lower in the list.
+
+However, in other technologies, more complex [merging] logic may apply, that effectively combines information from different representations of one verifier.
+
+Additionally, the concept of [masking] applies for all technologies.
+
+#### Overriding
+
+In technologies that use overriding, verifier files shadow other variants of the same verifier based on filename equality.
+
+For example, `/etc/voa/fedora/packages/default/ssh/buildserver1.pub` overrides `/usr/share/voa/fedora/package/default/ssh/buildserver1.pub`.
+
+Overriding depends on file name equality and directory structure.
+
+For example, `/etc/voa/fedora/packages/default/ssh/buildserver2.pub` does not override `/usr/share/voa/fedora/package/default/ssh/buildserver1.pub`, but instead adds an additional _signature verifier_.
+
+Great care has to be taken to ensure the naming scheme has the desired overriding behavior (e.g. to make sure that a revoked version of an SSH public key overrides a non-revoked version of the same SSH public key found in a different load path).
+
+#### Merging
+
+Some technologies use a business logic that "merges" different versions of the same _signature verifier_ (instead of applying overriding semantics).
+The purpose is to use the latest available information about each verifier, combining the information from all sources.
+
+When merging, information about a verifier found anywhere in the current _verifier lookup_ must be applied for all uses of a verifier (e.g. for revocation status).
+
+Wherever it is applicable and feasible, VOA technology backend implementations should implement "merging" to obtain a unified view of the latest information for each verifier.
+
+For example, if a verifier has been revoked, but only one of two copies of that verifier in VOA reflects this revocation, "merging" the two leads to stable visibility of the revocation status.
+Summarized, in technologies that implement merging, [load path] priority is not relevant (e.g. a non-revoked version in `/etc/voa/` of the same verifier never overrides the revoked version in `/usr/share/voa/`).
+The "latest normalized view" of a verifier is not impacted by load path priority.
+
+### Future compatibility
+
+No changes to the overall VOA structure are anticipated, which is why versioning is not encoded in the directory structure.
+However, if it turns out that a breaking change to the VOA specification is necessary, either a separate directory structure (e.g. `/usr/share/voa2` etc.), or a new subdirectory such as `/usr/share/voa/v2` can be specified.
```

Comment:
Adding `v2` to the VOA path, would break the assumption that `$os` should follow the root, for instance implementations would not be able to assume `os` lives under the root, and would cause multiple implementations/checks "if v2 exists" in: `/usr/share/voa/$os/$purpose/$context/$technology/`.

Perhaps it would be wise to have the version number in the path from the get-go? If we anticipate/recommend doing it for v2 already in this description :)